File: backend/schemas/customer.py
"""
OrderHub CRM — Customer Schemas
"""

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, EmailStr, Field


class CustomerCreate(BaseModel):
    email: EmailStr
    full_name: str = Field(..., max_length=255)
    country: str | None = Field(None, max_length=2)


class CustomerResponse(BaseModel):
    id: uuid.UUID
    email: str
    full_name: str
    country: str | None
    phone: str | None = None
    shipping_city: str | None = None
    shipping_city_ref: str | None = None
    shipping_warehouse_ref: str | None = None
    created_at: datetime
    updated_at: datetime
    
    order_count: int = 0  # computed field

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/customer.ts
import type { PaginatedResponse } from './common';

export interface CustomerListItem {
  id: string;
  email: string;
  full_name: string;
  country: string | null;
  phone?: string | null;
  shipping_city?: string | null;
  shipping_city_ref?: string | null;
  shipping_warehouse_ref?: string | null;
  created_at: string;
  updated_at: string;
  order_count: number;
}

export interface CustomerListFilters {
  page?: number;
  limit?: number;
  search?: string;
}

export type CustomerListResponse = PaginatedResponse<CustomerListItem>;
